logging/logging.go

```diff
@@ -76,12 +75,12 @@ func NewTestLogger(tb testing.TB) Logger {
 
 // NewObservedTestLogger is like NewTestLogger but also saves logs to an in memory observer.
 func NewObservedTestLogger(tb testing.TB) (Logger, *observer.ObservedLogs) {
-	logger := zaptest.NewLogger(tb, zaptest.WrapOptions(zap.AddCaller()))
+	logger := NewViamLogger("")
+	logger.AddAppender(NewStdoutAppender())
 	observerCore, observedLogs := observer.New(zap.LevelEnablerFunc(zapcore.DebugLevel.Enabled))
```

Comment:
An `observerCore` [is a](https://github.com/uber-go/zap/blob/master/zapcore/core.go#L25-L45) `zapcore.Core`.

Notably it [satisfies](https://github.com/uber-go/zap/blob/master/zapcore/core.go#L40-L44) the [Appender interface](https://github.com/viamrobotics/rdk/blob/8356443673d4b7be9c34edf935981b4df19b82af/logging/impl.go#L16-L22) for writing log entries. And writing this comment I realized our `Appender` interface is totally in the wrong file.

The `observedLogs` is just a [slice of log entries](https://github.com/uber-go/zap/blob/87577d85d58b6d92d0158967df29303d04d30e36/zaptest/observer/observer.go#L36-L40) that tests can inspect.